### PR TITLE
JWT specifications for Wings API.

### PR DESCRIPTION
**Pull request by real2two**
These are the JWT specification for Wings API.

### CODE REVIEW

**Comment by real2two** [comment]
I finished this, unless you want me to format this in a different way.

**Comment by devnote-dev** [comment]
This isn't really what I had in mind for JWT specifications – most of what you've written has already been stated in some parts of the documentation, for example the [public Wings endpoints](https://github.com/devnote-dev/ptero-notes/blob/8f8c6ae86d6c3445faf68d5e55f60953daac4bb0/wings/public.md). The specifications page would be more towards the actual construction of the tokens and the specific headers and claims needed for them (much like what you've put in the notes section). If you could change it to be like this, that would be great.

**Comment by real2two** [comment]
How's this?

File: wings/README.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The Wings API is used for communicating to the panel and other Wings instances o
 > **Note**
 > The Wings API does not have any control over the literal server instances in the panel, only the Docker containers for said servers and the server volumes. Use the [application API](/pterodactyl/application/README.md) for server control.
 
-# Access
+## Access
 
 The `Authorization` header must be present in the requests and must be prefixed with "Bearer " followed by the node's token (also known as the daemon token or master daemon key). The token can be obtained in the configuration page of a node in the admin panel, or from the [node configuration endpoint](/pterodactyl/application/nodes.md#get-nodesidconfiguration) in the application API. The `Accept` header must be set to "application/json" for requests, and the `Content-Type` must be set appropriately when making `POST`, `PATCH` or `PUT` requests.
+
+[Click here for JWT specifications for Wings API.](jwt.md)
```
Comment (devnote-dev) [inline wings/README.md]: This change isn't needed.
```suggestion
```

File: wings/jwt.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Wings API - JWT specifications
```
Comment (devnote-dev) [inline wings/jwt.md]:
```suggestion
# JWT specifications
```

File: wings/jwt.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- Tokens typically expire in 5 minutes with few exceptions.
+- JWTs use the `HS256` algorithm.
+
+## Payload format
+
+**Make sure to include the other required values based on the route the JWT is for.**
+
+| Field      | Type     | Description                                                                                                            |
+| ---------- | -------- | ---------------------------------------------------------------------------------------------------------------------- |
+| iss        | string   | The value for `remote` in config.json. This is the panel's url.                                                        |
```
Comment (devnote-dev) [inline wings/jwt.md]: Is this meant to be `config.yml`? I'm not sure where `config.json` is from.

File: wings/jwt.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+| aud        | string[] | An array containing a value of the the FQDN value, which is the Wings url.                                             |
+| unique_id  | string   | A random value. This can be anything.                                                                                  |
+| jti        | string   | MD5 encryption of either the value of `server_uuid` (in most cases) or `sub` (for `POST /api/servers/:uuid/transfer`). |
+
+Don't forget that most tokens require `server_uuid` in the JWT as well with the exception of `POST /api/servers/:uuid/transfer`, which uses `sub` instead.
+
+### Example JWT payload
+
+[This JWT comes from `GET /download/file`.](https://github.com/devnote-dev/ptero-notes/blob/8f8c6ae86d6c3445faf68d5e55f60953daac4bb0/wings/public.md#get-downloadfile)
```
Comment (devnote-dev) [inline wings/jwt.md]:
```suggestion
This JWT comes from [`GET /download/file`.](https://github.com/devnote-dev/ptero-notes/blob/8f8c6ae86d6c3445faf68d5e55f60953daac4bb0/wings/public.md#get-downloadfile)
```

File: wings/jwt.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+Example JWT value: `eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiIsImp0aSI6IjFhMjM0NTY3ODliMGNkZTEyM2Y0NTZnNzhoOTAxMjM0In0.eyJpc3MiOiJodHRwOi8vMTI3LjAuMC4xIiwiYXVkIjpbImh0dHA6Ly8xMjcuMC4wLjE6ODA4MCJdLCJqdGkiOiIxYTIzNDU2Nzg5YjBjZGUxMjNmNDU2Zzc4aDkwMTIzNCIsImlhdCI6MTY4Njc5NzQzNCwibmJmIjoxNjg2Nzk3MTM0LCJleHAiOjE2ODY3OTgzMzQsImZpbGVfcGF0aCI6Ii9zZXJ2ZXIucHJvcGVydGllcyIsInNlcnZlcl91dWlkIjoiMTIzNGE1NjctODliMC0xY2QyLTM0NTYtNzg5MGVmMTIzNDVnIiwidXNlcl91dWlkIjoiMWEyYmMzNGQtNTY3ZS04ZjkwLWcxaGktMjM0ajU2a2w3OG1uIiwidXNlcl9pZCI6MSwidW5pcXVlX2lkIjoiQ3lwZmZMcDB3RTBXdjAwNCJ9.zQkWBDuk05cuKZlmE5mfCdJSCZ4TY48siRFXZYLkpjY`
```
Comment (devnote-dev) [inline wings/jwt.md]:
````suggestion
```eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiIsImp0aSI6IjFhMjM0NTY3ODliMGNkZTEyM2Y0NTZnNzhoOTAxMjM0In0.eyJpc3MiOiJodHRwOi8vMTI3LjAuMC4xIiwiYXVkIjpbImh0dHA6Ly8xMjcuMC4wLjE6ODA4MCJdLCJqdGkiOiIxYTIzNDU2Nzg5YjBjZGUxMjNmNDU2Zzc4aDkwMTIzNCIsImlhdCI6MTY4Njc5NzQzNCwibmJmIjoxNjg2Nzk3MTM0LCJleHAiOjE2ODY3OTgzMzQsImZpbGVfcGF0aCI6Ii9zZXJ2ZXIucHJvcGVydGllcyIsInNlcnZlcl91dWlkIjoiMTIzNGE1NjctODliMC0xY2QyLTM0NTYtNzg5MGVmMTIzNDVnIiwidXNlcl91dWlkIjoiMWEyYmMzNGQtNTY3ZS04ZjkwLWcxaGktMjM0ajU2a2w3OG1uIiwidXNlcl9pZCI6MSwidW5pcXVlX2lkIjoiQ3lwZmZMcDB3RTBXdjAwNCJ9.zQkWBDuk05cuKZlmE5mfCdJSCZ4TY48siRFXZYLkpjY
```
````

File: wings/jwt.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+Header
```
Comment (devnote-dev) [inline wings/jwt.md]:
```suggestion
**Header**
```

File: wings/jwt.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+```json
+{
+  "typ": "JWT",
+  "alg": "HS256",
+  "jti": "1a23456789b0cde123f456g78h901234"
+}
+```
+
+Payload
```
Comment (devnote-dev) [inline wings/jwt.md]:
```suggestion
**Payload**
```

**Review by devnote-dev — CHANGES_REQUESTED** [review]
Looking good, just a few things.

File: wings/jwt.md (hunk 3 above)
Comment (real2two) [reply wings/jwt.md]: Thanks for pointing it out. I meant `config.yml`. I'll fix the text later.

Comment (real2two) [reply wings/jwt.md]: Fixed!

**Comment by real2two** [comment]
I fixed/commited the requested changes.

**Review by devnote-dev — APPROVED** [review]

**Pull request merged**